src/lcm/dispatchers.py

```diff
@@ -1,24 +1,34 @@
 import inspect
+from collections.abc import Callable
+from typing import TypeVar
 
 from jax import vmap
 
 from lcm.functools import allow_args, allow_kwargs
 
+F = TypeVar("F", bound=Callable)
 
-def spacemap(func, dense_vars, sparse_vars, *, dense_first):
+
+def spacemap(
+    func: F,
+    dense_vars: list[str],
+    sparse_vars: list[str],
+    *,
+    dense_first: bool,
```

Comment:
```suggestion
    put_dense_first: bool,
```
Makes it clear that this is about outputs.